dimod/include/dimod/constrained_quadratic_model.h

```diff
@@ -0,0 +1,493 @@
+// Copyright 2022 D-Wave Systems Inc.
```

Comment:
I see lots of asserts, but for this cqm, where performance is less important should we have std::throw insstead of asserts ?

Reply:
The division of labour here is at the C++ level we use asserts for performance, but we test everything and throw at the Python/Cython level.